sympy/core/relational.py

```diff
@@ -216,19 +216,25 @@ def canonical(self):
         elif tuple(ordered(args)) != args:
             r = r.reversed
 
+        LHS_CEMS = getattr(r.lhs, 'could_extract_minus_sign', None)
+        RHS_CEMS = getattr(r.rhs, 'could_extract_minus_sign', None)
+
+        if isinstance(r.lhs, BooleanAtom) or isinstance(r.rhs, BooleanAtom):
+            return r
```

Comment:
Why do we want so special case BooleanAtom here?

Am I right in thinking that this is why we have this? (on master)
```julia
In [4]: Eq(True, x).canonical                                                                                                     
Out[4]: x = True

In [5]: Eq(x, True).canonical                                                                                                     
Out[5]: x = True
```